src/controllers/cohort.js

```diff
@@ -5,11 +5,26 @@ import ERR from '../utils/errors.js'
 import { dateRegex } from '../utils/regexMatchers.js'
 
 export const create = async (req, res) => {
-  const { startDate, endDate } = req.body
+  const { name, course, startDate, endDate } = req.body
 
-  if (!startDate || !endDate) {
+  const foundCohorts = await getAllCohorts()
+
+  const courseInUse = foundCohorts.some((cohort) =>
+    cohort.course
+      .split(' ')
+      .join('')
+      .toLowerCase()
+      .includes(course.split(' ').join('').toLowerCase())
+  )
```

Comment:
If we have 100 cohorts and 5 teachers decide to create a new cohort, this will load 500 cohort objects into memory and then you'll be linearly looping over all of them to perform this `some` check. This is probably really inefficient.

I think we should do the filtering at the database level if it needs to be done. Have a prisma query that finds cohorts by course, it's much more efficient to do it at a DB level than in JS.

Reply:
For sure, will change it indeed!